boards/arm/nrf9160_innblue21/innblue21_board_init.c

```diff
@@ -31,15 +31,16 @@ static int pwr_ctrl_init(const struct device *dev)
 	const struct device *gpio;
 
 	/* Get handle of the GPIO device. */
-	gpio = device_get_binding(DT_LABEL(DT_NODELABEL(gpio0)));
+	gpio = DEVICE_DT_GET(DT_NODELABEL(gpio0));
 
-	/* Valid handle? */
-	if (gpio != NULL) {
-		/* Configure the gpio pin. */
-		config_pin(gpio, VDD_3V3_PWR_CTRL_GPIO_PIN);
-		config_pin(gpio, VDD_5V0_PWR_CTRL_GPIO_PIN);
+	if (!device_is_ready(gpio)) {
+		return -ENODEV;
 	}
 
+	/* Configure the gpio pin. */
+	config_pin(gpio, VDD_3V3_PWR_CTRL_GPIO_PIN);
+	config_pin(gpio, VDD_5V0_PWR_CTRL_GPIO_PIN);
```

Comment:
even though it's good to get rid of `DT_LABEL`, I think it's better to take the opportunity to also use `gpio_dt_spec` and Devicetree to describe this information, or, regulators in some cases like this.

examples:
https://github.com/zephyrproject-rtos/zephyr/pull/43817
https://github.com/zephyrproject-rtos/zephyr/pull/44082

Reply:
Using `gpio_dt_spec` means all hardware description is kept in Devicetree instead of hardcoded in C. Many boards also use custom board C code to enable power rails, for which we have the regulator API (or now power domains as well). In many cases, using the regulator API allows to delete custom init code entirely, and, allows for some optimizations like enable regulator conditionally (e.g., if it's powering a sensor, enable regulator subsystem only if sensors are enabled). See for example https://github.com/zephyrproject-rtos/zephyr/pull/43820

Reply:
@gmarull I see your point, but isn't that improvement out of scope for this PR? I think we can merge this improvement and make further improvements later if someone has time. This is a nice standalone change that accomplishes its purpose.

Reply:
Fair enough, btw, I've opened https://github.com/zephyrproject-rtos/zephyr/pull/46921